var/spack/repos/builtin/packages/py-dipy/package.py

```diff
@@ -19,9 +19,11 @@ class PyDipy(PythonPackage):
     homepage = "https://dipy.org/"
     pypi = "dipy/dipy-1.4.1.tar.gz"
 
+    version("1.7.0", sha256="59bb647128aae7793215c813bb8ea35dae260ac9f0d938c724064f0af5a05cc3")
     version("1.4.1", sha256="b4bf830feae751f3f985d54cb71031fc35cea612838320f1f74246692b8a3cc0")
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@:3.9", type=("build", "run"), when="@:1.4")
```

Comment:
optional:
switch the order of the `type` and `when` argument as most of the other python packages do